kn/plugin/event.rb

```diff
@@ -0,0 +1,48 @@
+require_relative '../os'
+require_relative '../checksums'
+
+module Kn
+module Plugin
+
+class Event
+  attr_reader :version, :source_bin, :target_bin
+
+  def initialize(version)
+    @version = version
+    
+    @os = if Kn::OS.mac? then 'darwin' else 'linux' end
+    @arch = 'amd64'
```

Comment:
Shouldn't it be determined by specific host architecture similarly to OS?

Reply:
If we'd like, we can refactor other scripts in this manner and add support to more os-arch pairs.